opendg/data/data.py

```diff
@@ -0,0 +1,328 @@
+            edge_dict = self._load_csv(file_path)
+        else:
+            edge_dict = data
+
+        #* check if edge_dict is sorted
+        t_list = list(edge_dict.keys())
+        if (not all(t_list[i] <= t_list[i+1] for i in range(len(t_list) - 1))):
+            edge_dict = dict(sorted(edge_dict.items()))
+
+        #! add logic to insert node events too
+        self.__dict__['_store']['events'] = self._to_eventStore(edge_dict)
+        ts_list = list(edge_dict.keys())
+        self.__dict__['_store']['min_time'] = ts_list[0]
+        self.__dict__['_store']['max_time'] = ts_list[-1]
+        self.__dict__['_store']['timestamps'] = ts_list
+        
+
+    def _to_eventStore(self, edge_dict:Dict[int,list]) -> Dict[int, EventStore]:
+        r"""Converts a dictionary of edges to a dictionary of events."""
+        event_dict = {}
+        for t, edges in edge_dict.items():
+            edge_index = torch.tensor(edges)
+            edge_index = torch.transpose(edge_index, 0, 1)
+            event_dict[t] = EventStore(timestamp=t,edges=edge_index,node_feats=None)
+        return event_dict
+
+    def _load_csv(self, file_path: str) -> Dict[str, Any]:
+        r"""Loads a continuous time dynamic graph from a csv file."""
+        os.path.exists(file_path), f"File {file_path} does not exist"
+        edge_dict = {}
+        with open('test_ctdg.csv', mode='r') as ctdg_file:
+            ctdg_reader = csv.reader(ctdg_file)
+            header = next(ctdg_reader)
+            assert len(header) >= 3, "The csv file should have at least 3 columns (source, destination, timestamp)"
+            #* infer the headers
+            header_dict = {header[i]:i for i in range(len(header))}
+            src_idx = header_dict.get('source',None)
+            dst_idx = header_dict.get('destination',None)
+            time_idx = header_dict.get('timestamp',None)
+
+            for edge in ctdg_reader:
+                u = int(edge[src_idx])
+                v = int(edge[dst_idx])
+                t = int(edge[time_idx])
+                if t not in edge_dict:
+                    edge_dict[t] = [(u,v)]
+                else:
+                    edge_dict[t].append((u,v))
+        return edge_dict
+
+
+
+    def __getattr__(self, key: str) -> Any:
+        return getattr(self._store, key)
+    
+    def __setattr__(self, key: str, value: Any):
+        setattr(self._store, key, value)
+
+    def __delattr__(self, key: str):
+        delattr(self._store, key)
+
+    def __getitem__(self, key: str) -> Any:
+        return self._store[key]
+
+    def __setitem__(self, key: str, value: Any):
+        self._store[key] = value
+
+    def __delitem__(self, key: str):
+        if key in self._store:
+            del self._store[key]
+
+    def __copy__(self):
+        return CTDG(self._store.copy())
+
+    def __deepcopy__(self, memo):
+        return CTDG(self._store.copy())
+
+    def __repr__(self) -> str:
+        return f"CTDG({self._store.keys()})"

```

Comment:
I'm unsure how helpful this `__repr__` is for debugging, and since we don't override `__str__` it's what the user will see if they stringify the object. If we have thousands of edges, this will just bombard their terminals.

I don't really have a good solution, but we may look into how `numpy` does their `__repr__` with large arrays, for instance: https://github.com/numpy/numpy/blob/v2.1.0/numpy/_core/arrayprint.py#L1609-L1656

In short, among other things, they will compactify large arrays like this:
```python
>>> repr(np.array([i for i in range(1001)]))
'array([   0,    1,    2, ...,  998,  999, 1000])'
```